app/forms/collection_form.rb

```diff
@@ -11,15 +11,25 @@ class CollectionForm < Reform::Form
   property :access, default: 'world'
   property :creator, writable: false
   property :depositor_sunets, virtual: true, prepopulator: lambda { |_options|
-    self.depositor_sunets = model.depositors.map { |user| user.email.delete_suffix('@stanford.edu') }
+    self.depositor_sunets = depositor_sunets_from_model.join(', ')
   }
 
   def sync(*)
     sunetids = depositor_sunets.split(/\s*,\s*/)
     emails = sunetids.map { |sunet| "#{sunet}@stanford.edu" }
-    model.depositors = emails.map { |email| User.create_or_find_by(email: email) }
+    model.depositors = emails.map do |email|
+      # It's odd that we need to do both, but this is how it's written.
+      # See: https://github.com/rails/rails/issues/36027
+      User.find_by(email: email) || User.create_or_find_by(email: email)
```

Comment:
I don't grok why we need a find as part of the second operation. Why not:

```ruby
User.find_by(email: email) || User.create(email: email)
```

Reply:
Your suggestion leaves the possibility of a race condition where the find fails, but another thread has created the user before our code gets to create.

Reply:
OK, good catch. mind dropping a comment about this incantation being used to avoid said race condition?